src/hooks/tables.tsx

```diff
@@ -66,3 +66,33 @@ export function useGetTable({ catalog, schema, table }: GetTableParams) {
     },
   });
 }
+
+export interface DeleteTableMutationParams extends Pick<TableInterface, 'catalog_name' | 'schema_name' | 'name'> {}
+
+interface DeleteTableParams {
+  onSuccessCallback?: () => void;
+}
+
+export function useDeleteTable({ onSuccessCallback }: DeleteTableParams = {}) {
+  const queryClient = useQueryClient();
+
+  return useMutation<void, unknown, DeleteTableMutationParams>({
+    mutationFn: async (params: DeleteTableMutationParams) : Promise<void> => {
+      const response = await fetch(`${UC_API_PREFIX}/tables/${params.catalog_name}.${params.schema_name}.${params.name}`, {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+      });
+      if (!response.ok) {
+        throw new Error('Failed to delete table');
+      }
+
+      return;
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['listTables'] });
```

Comment:
can you add catalog and schema to queryKey so we just invalidate this specific query?